Features/Shop.feature

```diff
@@ -2,73 +2,48 @@
 
 Feature: Shop
 
-As a User, I want to group products into tabs,
-I can filter products by values on those tabs,
-I can add items to the cart, 
+As a User, 
+I want to filter products by values,
+I should br able to add items to the cart, 
 and remove added items from the cart.
 
-Scenario: Verify every Tab <h1>:
-
-	#Arrange:
-	Given I am on 'shop' page
-	#Act:
-	When I click on '<Tab Name>' tab
-	#Assert:
-	Then I see '<h1>' heading is displayed
-
-Examples:
-	| Tab Name           | h1                |
-	| Сонячні панелі     | Сонячні панелі    |
-	| Інвертори          | Сонячні інвертори |
-	| Акумулятори        | Акумулятори       |
-	| Контролери заряду  | Контролери заряду |
-	| Системи кріплення  | Системи кріплення |
-	| Кабель і комутація | Сонячний кабель   |
-
-
 #Verify that catalog filter works correctly
 
-Scenario: Verify Filter on the tab:
+Scenario: Filter:
 
 	#Arrange:
-	Given I am on 'shop/<Tab>' page
+	Given I am on '<Tab>' page
 	#Act:
-	When I click on Filter button
-	And I click on '<FilterValue>' checkbox
+	When I see '<Page Heading>'
```

Comment:
I looked into implementation and as I expected, "Act" happens inside "Assert". Probably because you don't know how to transfer "itemsBefore" and "itemsAfter" values from When to Then step. 
We have an example of such transfer for accountPresetup variable in
https://github.com/OleksiiKhorunzhak/hilel-aqa/blob/main/AtataSpecFlow/AtataSamples.SpecFlow/Features/Account/AccountSteps.cs
So you need to add constructor to the steps class, like
 public AccountSteps(FeatureContext featureContext)
        {
            _featureContext = featureContext;
        }
and use _featureContext  as a dictionary to save 
featureContext["countBefore"] = itemsBefore
and read
var itemsBefore= featureContext.Get<int>("countBefore");;

Reply:
thank foк comment - it was the problem for me to resolve this

Reply:
fixed